vpc/tool/container/setup_container_linux.go

```diff
@@ -144,9 +174,11 @@ func configureLink(ctx context.Context, nsHandle *netlink.Handle, link netlink.L
 		}
 	}
 
-	err = addIPv4AddressAndRoute(ctx, nsHandle, link, allocation.IPV4Address)
-	if err != nil {
-		return err
+	if !withTrans || (withTrans && isTransLink) {
```

Comment:
This is hard for me to understand.

Could this logic be simplified if these toggles were called `enableIPv4` and `enableIPv6` and let the caller figure out exactly what to pass through, given the transition logic?